pkg/provider/cloud.go

```diff
@@ -28,24 +28,18 @@ type cloud struct {
 }
 
 type CloudConfig struct {
-	Kubeconfig   string             `yaml:"kubeconfig"` // The kubeconfig used to connect to the underkube
-	LoadBalancer LoadBalancerConfig `yaml:"loadbalancer"`
-	Instances    InstancesConfig    `yaml:"instances"`
-	Zones        ZonesConfig        `yaml:"zones"`
+	InfraKubeconfig string             `yaml:"infraKubeconfig"` // The kubeconfig used to connect to the infra kubevirt cluster
+	LoadBalancer    LoadBalancerConfig `yaml:"loadbalancer"`
+	InstancesV2     InstancesV2Config  `yaml:"instances"`
 }
 
 type LoadBalancerConfig struct {
 	Enabled              bool `yaml:"enabled"`              // Enables the loadbalancer interface of the CCM
 	CreationPollInterval int  `yaml:"creationPollInterval"` // How many seconds to wait for the loadbalancer creation
 }
 
-type InstancesConfig struct {
-	Enabled             bool `yaml:"enabled"`             // Enables the instances interface of the CCM
-	EnableInstanceTypes bool `yaml:"enableInstanceTypes"` // Enables 'flavor' annotation to detect instance types
-}
-
-type ZonesConfig struct {
-	Enabled bool `yaml:"enabled"` // Enables the zones interface of the CCM
+type InstancesV2Config struct {
+	Enabled bool `yaml:"enabled"` // Enables the instances interface of the CCM
```

Comment:
I am not sure if we need to keep the config to enable instance controller, I guess we always want to run it, the same for Load Balancer.

Reply:
So it turned out that for LoadBalancer it makes to keep the `Enabled` option. Let's just keep all options as they are.